File: python-euler/problem3.py
'''
The prime factors of 13195 are 5, 7, 13 and 29.

What is the largest prime factor of the number 600851475143?

==> 6857
'''
from toolkit import *

factors = factor(600851475143)
max = 0
for factor in factors:
    if isPrime(factor) and factor > max:
        max = factor

print max

File: java-euler/src/solutions/Problem3.java
package solutions;

import java.util.ArrayList;

/*
 * Problem:
 * 	The prime factors of 13195 are 5, 7, 13 and 29.
 * 	What is the largest prime factor of the number 600851475143?
 * 
 * Answer: 6857
 */
public class Problem3 {

	public static void main(String[] args) {
		long max = 0;
		ArrayList<Long> factors = Helpers.factor(600851475143L);
		
		for (long factor: factors) {
			if (Helpers.isPrime(factor) && factor > max) {
				max = factor;
			}
		}
		
		System.out.println(max);
	}

}